controllers/datafoundation.go

```diff
@@ -329,3 +334,29 @@ func (r *dataFoundationReconciler) reconcileRookCephOperatorConfig() error {
 
 	return nil
 }
+
+func (r *dataFoundationReconciler) reconcileOCSInitialization() error {
+	r.Log.Info("Reconciling OCSInitialization")
+
+	ocsInitList := ocsv1.OCSInitializationList{}
+	if err := r.list(&ocsInitList, client.InNamespace(r.dfNamespace)); err != nil {
+		return fmt.Errorf("failed to list OCSInitialization resources: %v", err)
+	}
+	if len(ocsInitList.Items) == 0 {
+		r.Log.V(-1).Info("OCSInitialization resource not found")
+		return nil
+	}
+	obj := &ocsInitList.Items[0]
+	err := r.GetAndUpdate(obj, func() error {
```

Comment:
- why is there a `GetAndUpdate` function in the first place
- what info are we missing in the supplied `obj` that needs explicit `get` again?

Reply:
> why is there a GetAndUpdate function in the first place
Because there are few reconcile functions where we don't want to update when the obj is in the desired state.
I have updated the code so I'm not explicitly getting the object again.

Reply:
@leelavg GetAndUpdate should be the recommended way for this point forward to get an object, mutate it using a mutate function, assert changes (if any), and update the object (if needed). 

This is the result of us observing almost the same pattern repeating itself in multipole reconcile functions.
And from code structure perspective we followed the `CreateAndUpdate` pattern (without the create part)

Reply:
> there are few reconcile functions where we don't want to update when the obj is in the desired state.
- ack, I thought sdk already checks the equality before update